react/homework.md

```diff
@@ -0,0 +1,135 @@
+
+The homework should be done in one codesandbox. Below are the steps with detailed requirements.
+
+  **The usage of Redux or any other solutions for state management is prohibited.**
+
+  **The usage of any libraries with components or form libraries is prohibited.** 
+
+## Steps
+
+1. Create a new app using the React template on https://codesandbox.io
+2. Use function components with hooks.
+3. Draw the Search Bar and Cards in the div tag with id "search-bar".
+4. Input value should be saved to LocalStorage during component’s unmount. During the initialization pick the value from LocalStorage and show it.
+
+All logical parts should be set into separate components.
+Try to make it simple.
+
+* examples of Search Bar:
+![example of Search Bar](images/BlogHeader_SearchBar_1200x600.png)
+![example of Search Bar](images/cssheader1.webp)
+
+* examples of Cards:
+![example of Cards](images/behance.jpeg)
+![example of Cards](images/rightmove.jpeg)
+![example of Cards](images/awwwards.jpeg)
+
+4. Create a component with form in a new div tag with id "form-component".
+5. Form should be implemented using uncontrolled components.
+6. Add functionality to your app:
+   +  Collect information through a form
+   +  You can collect any type of information, but form has to include at least one control of the following types:
+      + text input
+      + date input
+      + dropdown/select
+      + checkbox
+      + switcher
+      + file upload (image)
+
+   **Example:**
+     ```
+     text input - name, surname, zip-code;
+     date input - birthday, date of delivery;
+     dropdown/select - list of countries, list of states (User can choose only one element from the list)
+     checkbox - "I consent to my personal data" field, list of extra presents (User can choose several items from the list)
+     switcher - male/female, "I want to receive notifications about promo, sales, etc." / "I don’t want to receive notifications about promo, sales, etc."
+     file upload - profile picture
+     ```
+
+   After the Submit button is clicked, all data from the form has to be displayed as Card in the cards list below the form. The confirmation message (with information that the data has been saved) should be displayed, after that form has to be cleared.
+   
+   The quantity of cards should equal the number of form submissions (if the user filled in and submitted the form five times, then five cards have to be displayed under the form).
+
+   ### **Form validation**
+   
+   Validation should happen after the **Submit** button was clicked. In case some field was filled incorrectly, then error messages should be displayed under it. The card mustn’t be created until the user fixes all errors.
+   
+   After changing the value of erroneous input, the reset should happen only for this particular field, but not for the others.
+   
+   Submit button should be disabled at initialization (before the first typing), and until the user fixes all errors.
```

Comment:
can we create one more html file with examples of html elements and their designs in different states? buttons, inputs, download indicator